### ISSUE TITLE
False positive on `ComposeModifierWithoutDefault` 

### ISSUE THREAD

**Issue by yogurtearl** [slackhq/compose-lints#408]
Consider:
```kotlin
fun Modifier.customBackground(
     foo: Foo,
): Modifier {
     // compute background based on Foo
    background(.... )
}
```

### Actual

`ComposeModifierWithoutDefault` triggers on custom modifier extension factories. 

### Expected

`ComposeModifierWithoutDefault` should ignore `Modifier` as a receiver, and only trigger on  `Modifier` as an actual listed param.
